crates/typst/src/model/heading.rs

```diff
@@ -42,12 +42,29 @@ use crate::util::{option_eq, NonZeroExt};
 /// # Syntax
 /// Headings have dedicated syntax: They can be created by starting a line with
 /// one or multiple equals signs, followed by a space. The number of equals
-/// signs determines the heading's logical nesting depth.
+/// signs determines the heading's logical nesting depth. The `{offset}` field
+/// can be set to configure the starting depth.
 #[elem(Locatable, Synthesize, Count, Show, Finalize, LocalName, Refable, Outlinable)]
 pub struct HeadingElem {
-    /// The logical nesting depth of the heading, starting from one.
+    /// The logical nesting depth of the heading, starting from *one*. This is
+    /// combined with `{offset}` to compute the actual `{level}`.
```

Comment:
This could use a bit more explanation, why this is useful separately from `level`. Also again stating that this is what the syntax sets.